Modules/xASL_module_ASL.m

```diff
@@ -156,14 +153,21 @@
 end
 
 
-%% F. Split ASL and M0 within the ASL time series
+%% E. Split ASL and M0 within the ASL time series
 % Run this when the images hasn't been touched yet
 % The first three states are here, because the first two are run only conditionally
 
 % NOTE THAT THIS PART IS BEING PHASED OUT, AND NOW PROCESSED IN
 % `xASL_bids_parseM0` (BIDS->Legacy conversion). We keep this here only for
 % backward compatibility, for images imported a long time ago
 if ~x.mutex.HasState(StateName{1}) && ~x.mutex.HasState(StateName{2}) && ~x.mutex.HasState(StateName{3})
+	% Initialize the DummyScan and M0 position fields - by default empty
```

Comment:
Why did you change the order of everything, instead of simply creating new subfunctions from the big subfunction that we already had below, that are called from the big subfunction, as well as from the new location?

Reply:
It's easy to check. I haven't changed anything but order. You can see the order change on lines 24-27.

For the mult-parameter check where I did more changes, I created a subfunction